### PR TITLE
Jbake migration

### PR DESCRIPTION
**Pull request by tarilabs**
This PR performs JBake and Bootstrap 5 migration and is based on:

- the heavy lifting work from Jason's PR https://github.com/kiegroup/drools-website/pull/173

but superseeds it by:

- incorporate infrastructure from optaplanner-website `main` at the time of writing
so to re-use most of the structural parts across websites
- incorporates already https://github.com/kiegroup/optaplanner-website/pull/425
- pickup fix https://github.com/kiegroup/optaplanner-website/pull/427
- borrows https://github.com/kiegroup/optaplanner-website/pull/414
- borrows https://github.com/kiegroup/optaplanner-website/pull/350
- and many other misc drools-website related fixes, as documented by additional commits

Specifically adding on top commits:

- c33095f..006f4ea

# demo

Screenshot 2021-10-17 at 21 26 50

As it can be noticed, structural elements are borrowed/same as from Optaplanner-website 🚀 

We will need (separately from these PRs) the infrastructure /CI work to update periodically website and publish it, but I would expect it will be similar as for optaplanner.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,15 @@ It solves all known pitfalls that can disrupt your development.
...
+# How to build with JBake
+
+To build the drools-website (before and after your changes):
+
+```
+$ mvn clean generate-resources
+$ firefox target/website/index.html
+```
+
 # How to build with Awestruct
```
Comment (yurloc) [inline README.md:20]: The rest of the Readme is now obsolete, isn't it?

**Review by yurloc — COMMENTED** [review]
Don't forget to clean up the old stuff, such as:
- `/code/*`
- `/community/*`
- `/Gemfile`
- etc.

File: README.md [hunk 2]
```diff
@@ -8,6 +8,15 @@ It solves all known pitfalls that can disrupt your development.
 It also describes all guidelines, tips and tricks.
 If you want your pull requests (or patches) to be merged into master, please respect those guidelines.
 
+# How to build with JBake
+
+To build the drools-website (before and after your changes):
+
+```
+$ mvn clean generate-resources
+$ firefox target/website/index.html
```
Comment (LightGuard) [inline README.md:17]: I'd really prefer we get away the idea of viewing the site as a file.

Please use the below instead:
```
mvn clean jbake:inline
firefox http://localhost:8820
```

File: assets/website/javascriptLibraries.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Javascript libraries
+====================
+
+Twitter bootstrap version: 3.1.1
+Downloaded from: http://twitter.github.com/bootstrap/
+License: MIT (which can be used by Apache License v2.0 projects)
+
+jQuery version: 1.11.0
+Downloaded from: http://jquery.com/
+License: MIT (which can be used by Apache License v2.0 projects)
```
Comment (LightGuard) [inline assets/website/javascriptLibraries.txt:10]: These are no longer the correct versions, nor are they being downloaded and served from this host.

File: templates/shared/menu.ftl [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                <li class="nav-item"><a class="nav-link" href="${content.rootpath}code/sourceCode.html">Source</a></li>
+                <li class="nav-item"><a class="nav-link" href="${content.rootpath}community/team.html">Team</a></li>
+                <li class="nav-item"><a class="nav-link" href="${content.rootpath}product/services.html">Services</a></li>
+                <li class="nav-item dropdown">
+                    <a class="nav-link dropdown-toggle" href="#" id="navbarKIE" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+                        KIE
+                    </a>
+                    <ul class="dropdown-menu" aria-labelledby="navbarKIE">
+                        <!-- <li><a class="dropdown-item" href="https://www.drools.org/" target="_blank">Drools</a></li> -->
+                        <li><a class="dropdown-item" href="https://kogito.kie.org/" target="_blank">Kogito</a></li>
```
Comment (LightGuard) [inline templates/shared/menu.ftl:41]: You're missing a link back to the kie blog and the main kie.org site

File: assets/website/javascriptLibraries.txt (hunk 3 above)
Comment (tarilabs) [reply assets/website/javascriptLibraries.txt:10]: updated reflecting from:

- https://github.com/kiegroup/optaplanner-website/pull/406

File: README.md (hunk 2 above)
Comment (tarilabs) [reply README.md]: good catch, I documented the options I use in e8dcab6

**Pull request merged**